promise.js

```diff
@@ -79,25 +90,35 @@ function createPool (opts) {
 
     query: function (sql, args) {
       return new Promise(function (resolve, reject) {
-        corePool.query(sql, args, function (err, rows, fields) {
+        var done = function (err, rows, fields) {
```

Comment:
For some DRY how about
```js
function queryDoneCb(resolve, reject) {
  return function(err, rows, fields) {
    if (err) {
      reject(err)
    } else {
      resolve([rows, fields])
    }
  }
}
```
That could be used in 4 places (query, promise) x (query, execute)
```js
var done = doneCb(resolve, reject)
```